### PR TITLE
RE: Source code highlighting Issue #8

### PR DESCRIPTION
**Pull request by kellewic**
I really needed this functionality so powered through figuring it out. I have no experience with Mediawiki so forgive me if there's a better way to do this.

The syntax I used for testing is:

`{{#github:php_sessions/sync_php_sessions.sh|kellewic/scripts|master|lang="bash"|line=|start='3'|highlight=3,7, 10}}`

The "inline" param is also supported. Values can be quoted or not (single or double).

### CODE REVIEW

File: src/GitHubParserHook.php [hunk 1]
```diff
@@ -43,14 +53,58 @@ private function setFields( ProcessingResult $result ) {
...
+	}
+
+	private function getRenderedContent(Parser $parser) {
 		$content = $this->getFileContent();
 
 		if ( $this->isMarkdownFile() ) {
 			$content = $this->renderAsMarkdown( $content );
 		}
+		else if ($this->lang !== "") {
+			if ( ExtensionRegistry::getInstance()->isLoaded( 'SyntaxHighlight' ) ) {
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: ExtensionRegistry? The tests are failing cause of this reference

https://travis-ci.org/JeroenDeDauw/GitHub/jobs/215355374

File: src/GitHubParserHook.php [hunk 2]
```diff
@@ -22,6 +23,15 @@ class GitHubParserHook implements HookHandler {
...
 	private $branchName;
 
+	// Parameters for SyntaxHighlight extension (formerly SyntaxHighlight_GeSHi)
+	private $lang;
+	private $line;
+	private $start;
+	private $highlight;
+	private $inline;
+
+	private static $syntaxHighlightFields = array('lang', 'line', 'start', 'highlight', 'inline');
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: You could use `[]`. Also note that there should be spaces between the brackets and values

File: src/GitHubParserHook.php [hunk 3]
```diff
@@ -43,14 +53,58 @@ private function setFields( ProcessingResult $result ) {
 		$this->fileName = $params['file']->getValue();
 		$this->repoName = $params['repo']->getValue();
 		$this->branchName = $params['branch']->getValue();
+
+		foreach ( self::$syntaxHighlightFields as $val ) {
+			if ( isset( $params[$val] ) ) {
+				$this->$val = $this->cleanField( $params[$val]->getValue() );
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: Eh. This kind of dynamic shit is really bad style (in general, loose from mediawiki or PHP). No need for that complexity

File: src/Setup.php [hunk 4]
```diff
@@ -122,11 +122,32 @@ public function getGitHubHookDefinition() {
...
+					'default' => '',
+					'message' => 'github-par-highlight',
+				),
+				'inline' => array(
+					'default' => null,
+					'message' => 'github-par-inline',
+				),
 			),
-			array( 'file', 'repo', 'branch' )
+			array( 'file', 'repo', 'branch', 'lang', 'line', 'start', 'highlight', 'inline')
```
Comment (JeroenDeDauw) [inline src/Setup.php]: space at the end

File: src/Setup.php [hunk 5]
```diff
@@ -122,11 +122,32 @@ public function getGitHubHookDefinition() {
...
+				),
+				'start' => array(
+					'default' => '1',
+					'message' => 'github-par-start',
+				),
+				'highlight' => array(
+					'default' => '',
+					'message' => 'github-par-highlight',
+				),
+				'inline' => array(
```
Comment (JeroenDeDauw) [inline src/Setup.php]:
```php
'type' => 'boolean',
```

File: i18n/qqq.json [hunk 6]
```diff
@@ -8,5 +8,10 @@
 	"github-desc": "{{desc|name=GitHub|url=https://www.mediawiki.org/wiki/Extension:GitHub}}",
 	"github-par-filename": "Label for the contents of the \"file name\" parameter\n{{Identical|Filename}}",
 	"github-par-branchname": "Label for the contents of the \"branch name\" parameter",
-	"github-par-reponame": "Label for the contents of the \"repository name\" parameter"
+	"github-par-reponame": "Label for the contents of the \"repository name\" parameter",
+	"github-par-lang": "Label for the contents of the \"lang\" parameter",
+	"github-par-line": "Label for the contents of the \"line\" parameter",
+	"github-par-start": "Label for the contents of the \"start\" parameter",
+	"github-par-highlight": "Label for the contents of the \"highlight\" parameter",
+	"github-par-inline": "Label for the contents of the \"inline\" parameter",
```
Comment (JeroenDeDauw) [inline i18n/qqq.json]: It's be nice to have the descriptions be useful to anyone that does not know the SyntaxHighlight extension

File: src/Setup.php [hunk 7]
```diff
@@ -122,11 +122,32 @@ public function getGitHubHookDefinition() {
...
+				'lang' => array(
+					'default' => '',
+					'message' => 'github-par-lang',
+				),
+				'line' => array(
+					'default' => null,
+					'message' => 'github-par-line',
+				),
+				'start' => array(
+					'default' => '1',
```
Comment (JeroenDeDauw) [inline src/Setup.php]:
```php
'type' => 'integer',
```

File: src/Setup.php [hunk 8]
```diff
@@ -122,11 +122,32 @@ public function getGitHubHookDefinition() {
 					'aliases' => 'branchname',
 					'message' => 'github-par-branchname',
 				),
+				'lang' => array(
+					'default' => '',
+					'message' => 'github-par-lang',
+				),
+				'line' => array(
+					'default' => null,
```
Comment (JeroenDeDauw) [inline src/Setup.php]: Maybe integer type? idk what this thing does so hard to tell

```php
'type' => 'integer',
```

File: src/GitHubParserHook.php [hunk 9]
```diff
@@ -22,6 +23,15 @@ class GitHubParserHook implements HookHandler {
 	private $repoName;
 	private $branchName;
 
+	// Parameters for SyntaxHighlight extension (formerly SyntaxHighlight_GeSHi)
+	private $lang;
+	private $line;
+	private $start;
+	private $highlight;
+	private $inline;
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: Please use descriptive variable names. `$programmingLanguage`, `$startingLineNumber`, etc

File: src/GitHubParserHook.php [hunk 10]
```diff
@@ -43,14 +53,58 @@ private function setFields( ProcessingResult $result ) {
...
+		}
 	}
 
-	private function getRenderedContent() {
+	private function cleanField( $val ) {
+		if ( $val !== null ) {
+			$val = trim( $val, "'\"" );
+		}
+		return $val;
+	}
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: Just drop this please

File: src/GitHubParserHook.php [hunk 11]
```diff
@@ -43,14 +53,58 @@ private function setFields( ProcessingResult $result ) {
...
+
+	private function getRenderedContent(Parser $parser) {
 		$content = $this->getFileContent();
 
 		if ( $this->isMarkdownFile() ) {
 			$content = $this->renderAsMarkdown( $content );
 		}
+		else if ($this->lang !== "") {
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: spaces

File: src/GitHubParserHook.php [hunk 12]
```diff
@@ -43,14 +53,58 @@ private function setFields( ProcessingResult $result ) {
...
 
-	private function getRenderedContent() {
+	private function cleanField( $val ) {
+		if ( $val !== null ) {
+			$val = trim( $val, "'\"" );
+		}
+		return $val;
+	}
+
+	private function getRenderedContent(Parser $parser) {
```
Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: spaces

File: src/GitHubParserHook.php (hunk 3 above)
Comment (kghbln) [inline src/GitHubParserHook.php]: > shit

This wording is a bit discouraging. Call it "magic" or so and everybody also knows what is meant from context.

Comment (JeroenDeDauw) [inline src/GitHubParserHook.php]: Time travel bunny pretends to not know what you are talking about

timebunny

File: i18n/qqq.json (hunk 6 above)
Comment (kellewic) [inline i18n/qqq.json]: Where do these get displayed?

Comment (JeroenDeDauw) [inline i18n/qqq.json]: Huh... now that I think about it I'm not sure if anything that uses this still exists. There used to be something like `#mapsdoc` on this page https://sandbox.semantic-mediawiki.org/wiki/Mapsdoc_-_leaflet but then for all declaratively defined parser hooks. I suspect this got ditched when migrating to https://github.com/JeroenDeDauw/ParserHooks and was never added again. Perhaps I'm forgetting about something though. @kghbln any idea?

So this is fine for now. However expanding the usage section of the README with descriptions of the new parameters would be good.

Comment (kghbln) [inline i18n/qqq.json]: These system messages got introduced with version [1.0.0](https://github.com/JeroenDeDauw/GitHub/blob/439ca62d6b6f9615f67f73873da138e6fcccb1be/src/Setup.php#L100) but I am not sure how the were supposed to integrate, i.e. the name of the parser function supposed to display the stuff.

The messages added to "qqq.json" are shown to the translators of the system messages to provide assistance and context in understanding what the programmer actually meant with them. If you introduce system messages here you also need to add them in "en.json" which is the source file for system messages.

File: src/GitHubParserHook.php (hunk 1 above)
Comment (kellewic) [inline src/GitHubParserHook.php]: This was an initial idea to support multiple syntax highlighters, but after looking around most of them aren't  maintained or aren't stable. Of those that are, they use the same config as the one Mediawiki downloads with - I took it out.

**Pull request merged**